proposals/csharp-9.0/native-integers.md

```diff
@@ -43,34 +43,39 @@ There is an identity conversion between compound types that differ by native int
 The tables below cover the conversions between special types.
 (The IL for each conversion includes the variants for `unchecked` and `checked` contexts if different.)
 
+General notes:
+- Most docs explain `conv.u` as conversion to __native unsigned integer__ and `conv.i` as conversion to __native integer__. But a more appropriate description is that `conv.u` is zero-extending and `conv.i` is sign-extending (if the most significant bit is `1`, then the conversion fills with `1`s).
+- Some non-obvious choices are guided by pre-existing semantics. For example, `sbyte->nint` uses `conv.i` and `byte->nint` uses `conv.u`, the same way that `sbyte->long` uses `conv.i8` and `byte->long` uses `conv.u8`. Also, `checked(void*->nint)` uses `conv.ovf.i.un` the same way that `checked(void*->long)` uses `conv.ovf.i8.un`.
+
 | Operand | Target | Conversion | IL |
 |:---:|:---:|:---:|:---:|
 | `object` | `nint` | Unboxing | `unbox` |
-| `void*` | `nint` | PointerToVoid | `conv.i` |
+| `void*` | `nint` | PointerToVoid | `conv.i`/`conv.ovf.i.un` |
 | `sbyte` | `nint` | ImplicitNumeric | `conv.i` |
 | `byte` | `nint` | ImplicitNumeric | `conv.u` |
 | `short` | `nint` | ImplicitNumeric | `conv.i` |
 | `ushort` | `nint` | ImplicitNumeric | `conv.u` |
 | `int` | `nint` | ImplicitNumeric | `conv.i` |
-| `uint` | `nint` | ExplicitNumeric | `conv.u` / `conv.ovf.u` |
+| `uint` | `nint` | ExplicitNumeric | `conv.u` / `conv.ovf.i.un` |
 | `long` | `nint` | ExplicitNumeric | `conv.i` / `conv.ovf.i` |
-| `ulong` | `nint` | ExplicitNumeric | `conv.i` / `conv.ovf.i` |
-| `char` | `nint` | ImplicitNumeric | `conv.i` |
+| `ulong` | `nint` | ExplicitNumeric | `conv.i` / `conv.ovf.i.un` |
+| `char` | `nint` | ImplicitNumeric | `conv.u` |
 | `float` | `nint` | ExplicitNumeric | `conv.i` / `conv.ovf.i` |
 | `double` | `nint` | ExplicitNumeric | `conv.i` / `conv.ovf.i` |
 | `decimal` | `nint` | ExplicitNumeric | `long decimal.op_Explicit(decimal) conv.i` / `... conv.ovf.i` |
 | `IntPtr` | `nint` | Identity | |
 | `UIntPtr` | `nint` | None | |
+| | | | |
 | `object` | `nuint` | Unboxing | `unbox` |
 | `void*` | `nuint` | PointerToVoid | `conv.u` |
```

Comment:
Similar question to https://github.com/dotnet/csharplang/pull/6059/files#r858012318

Why do we emit `conv.u` for `void* to nuint`?

Based on the other conversions, it looks like `void*` is treated as `unsigned` and therefore this should effectively be a "nop" (unless there is some weird stack evaluation requirement I'm missing).